pkg/client/cloudevents/grpcsource/util.go

```diff
@@ -86,6 +90,39 @@ func ToManifestWork(rb *openapi.ResourceBundle) (*workv1.ManifestWork, error) {
 	return work, nil
 }
 
+func ToLabelSearch(opts metav1.ListOptions) (string, bool, error) {
+	if len(opts.LabelSelector) == 0 {
+		return "", false, nil
+	}
+
+	labelSelector, err := labels.Parse(opts.LabelSelector)
+	if err != nil {
+		return "", false, fmt.Errorf("invalid labels selector %q: %v", opts.LabelSelector, err)
+	}
+
+	requirements, selectable := labelSelector.Requirements()
+	if !selectable {
+		return "", false, nil
+	}
+
+	labels := []string{}
+	for _, requirement := range requirements {
+		if requirement.Operator() != selection.Equals {
+			// TODO support more operators
```

Comment:
With https://github.com/openshift-online/maestro/pull/136 being merged, can we check what other operators are supported and allow list them as valid operators? 

/cc @clyang82

Reply:
@machi1990 now only the `DoesNotExist` cannot be supported (if this is expected, we need enhance the `ocm-sdk` as a follow up), I added some test case in the e2e `List works with gRPC source ManifestWork client`, and I also add the support for label selector in the watch, PTAL, thanks

Reply:
Thanks @skeeey 

>  now only the DoesNotExist cannot be supported (if this is expected, we need enhance the ocm-sdk as a follow up)

That's noted. Just out of curiosity, wouldn't his be the opposite of what's done for Exists? At least that's my expectation but maybe I am missing something.

> I added some test case in the e2e List works with gRPC source ManifestWork client, and I also add the support for label selector in the watch, PTAL, thanks

I am giving it a look now. Thanks

Reply:
The change lgtm. 
Once this is merged, I can test it against my client I am building.